caom2-artifact-resolvers/src/main/java/ca/nrc/cadc/caom2/artifact/resolvers/CadcResolver.java

```diff
@@ -0,0 +1,149 @@
+*  General Public License for           Générale Publique GNU Affero
+*  more details.                        pour plus de détails.
+*                                       
+*  You should have received             Vous devriez avoir reçu une
+*  a copy of the GNU Affero             copie de la Licence Générale
+*  General Public License along         Publique GNU Affero avec
+*  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+*  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+*                                       <http://www.gnu.org/licenses/>.
+*
+*
+************************************************************************
+*/
+
+
+package ca.nrc.cadc.caom2.artifact.resolvers;
+
+import ca.nrc.cadc.auth.AuthMethod;
+import ca.nrc.cadc.auth.AuthenticationUtil;
+import ca.nrc.cadc.caom2.artifact.resolvers.util.ResolverUtil;
+import ca.nrc.cadc.net.StorageResolver;
+import ca.nrc.cadc.net.Traceable;
+import ca.nrc.cadc.reg.Standards;
+import ca.nrc.cadc.reg.client.RegistryClient;
+import org.apache.log4j.Logger;
+
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+
+/**
+ * This class can convert an URI into a URL to the CADC storage system.
+ *
+ * @author adriand
+ */
+public class CadcResolver implements StorageResolver, Traceable {
+
+    private static final Logger log = Logger.getLogger(CadcResolver.class);
+    public String scheme = "cadc";
+    public static URI STORAGE_INVENTORY_URI;
+
+    static {
+        try {
+            STORAGE_INVENTORY_URI = new URI("ivo://cadc.nrc.ca/minoc");
+        } catch (URISyntaxException e) {
+            e.printStackTrace();
+        }
+    }
+
+    protected AuthMethod authMethod;
+
+    @Override
+    public URL toURL(URI uri) {
+
+        try {
+            ResolverUtil.validate(uri, scheme);
+            // check if authMethod has been set
+            AuthMethod am = this.authMethod;
+            if (am == null) {
+                am = AuthenticationUtil.getAuthMethod(AuthenticationUtil.getCurrentSubject());
+            }
+            if (am == null) {
+                am = AuthMethod.ANON;
+            }
+            URL url = this.toURL(getServiceURL(am), uri);
+            log.debug(uri + " --> " + url);
+            return url;
+        } catch (MalformedURLException ex) {
+            throw new RuntimeException("BUG", ex);
+        } catch (URISyntaxException bug) {
+            throw new RuntimeException("BUG - failed to create data web service URI", bug);
+        }
+    }
+
+    public URL getServiceURL(final AuthMethod am) throws URISyntaxException {
+        // Convenient for mocking
+        RegistryClient rc = new RegistryClient();
+        return rc.getServiceURL(new URI(STORAGE_INVENTORY_URI.toASCIIString()), Standards.SI_FILES, am);
+    }
+
+    protected URL toURL(URL serviceEndPointURL, URI artifactURI) throws MalformedURLException {
```

Comment:
is there a reason to make this separate and protected? I would just inline it (but see other comments about getServiceURL).